README.adoc

```diff
@@ -0,0 +1,247 @@
+
+Due to the constraints on the development resources (read as, _"it is just
+me"_), I needed to take some shortcuts to get this project going:
+
+- https://github.com/reactor/reactor-core/[Reactor Core] for Reactive Streams
+
+- https://github.com/reactor/reactor-netty[Reactor Netty] for HTTP requests
+
+- https://github.com/googleapis/google-api-java-client[Google APIs Client
+  Library]footnote:[This could have been replaced with a more lightweight
+  alternative, but given you have already been using Pub/Sub, it is highly
+  likely that you already sold your soul to some other Google Cloud services
+  too. Hence, no need to introduce an extra dependency.] for authentication
+
+- https://github.com/FasterXML/jackson-databind[Jackson]footnote:[https://github.com/googleapis/google-api-java-client[Google
+  APIs Client Library] already depends on this library.] for JSON serialization
+
+- http://micrometer.io/[Micrometer] for metrics
+
+== Getting Started
+
+You first need to add the `reactor-pubsub` artifact into your list of
+dependencies:
+
+```xml
+<dependency>
+    <groupId>com.vlkan</groupId>
+    <artifactId>reactor-pubsub</artifactId>
+    <version>${reactor-pubsub.version}</version>
+</dependency>
+```
+
+In order to instantiate a publisher and/or subscriber, you need to have certain
+basic components in place:
+
+- `ScheduledExecutorService` and `Scheduler` for scheduling asynchronous tasks
+- `PubsubAccessTokenCache` for authenticating HTTP requests
+- `MeterRegistry` for metrics
+- `HttpClient` for HTTP requests
+- `PubsubClient` for Pub/Sub HTTP requests
+
+```java
+// Create the executor that Reactor will subscribe on.
+BoundedScheduledThreadPoolExecutor executorService =
+        new BoundedScheduledThreadPoolExecutor(
+                100, new ScheduledThreadPoolExecutor(8));
+Scheduler scheduler = Schedulers.fromExecutorService(executorService);
+
+// Create the Pub/Sub access token cache.
+PubsubAccessTokenCache accessTokenCache = PubsubAccessTokenCache
+        .builder()
+        .setExecutorService(executorService)
+        .setCredentials("awesome-credentials")
+        .build();
+
+// Create a meter registry for metrics.
+MeterRegistry meterRegistry = new SimpleMeterRegistry();
+
+// Create the Reactor Netty HTTP client.
+HttpClient httpClient = HttpClient.create();
+
+// Create the Pub/Sub client.
+PubsubClient client = PubsubClient
+        .builder()
+        .setAccessTokenCache(accessTokenCache)
+        .setHttpClient(httpClient)
+        .setMeterRegistry(meterRegistry)
+        .build();
+```
+
+You can create a publisher and publish a message as follows:
+
+```java
+// Create the publisher.
+String projectName = "awesome-project";
+PubsubPublisherConfig publisherConfig = PubsubPublisherConfig
+        .builder()
+        .setProjectName(projectName)
+        .setTopicName("awesome-topic")
+        .build();
+PubsubPublisher publisher = PubsubPublisher
+        .builder()
+        .setClient(client)
```

Comment:
API tip: since it won't work without the `client`, it would be better if the API would be `client` centric:
```java
// Option 1
var publisher = client.newPublisherBuilder().setConfig(config).build();

// Option 2
var publisher = PubsubPublisher.build(client).setConfig(publisherConfig).build();
```

Otherwise, if somebody needs an instance of `PubsubPublisher`, they may struggle to find all necessary bits to do so.